dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -1002,6 +1002,15 @@ private WorkflowVersion addDockstoreYmlVersionToWorkflow(String repository, Stri
                     existingWorkflowVersion.setKernelImagePath(remoteWorkflowVersion.getKernelImagePath());
                     existingWorkflowVersion.setReadMePath(remoteWorkflowVersion.getReadMePath());
                     existingWorkflowVersion.setDescriptionAndDescriptionSource(remoteWorkflowVersion.getDescription(), remoteWorkflowVersion.getDescriptionSource());
+                    // this kinda sucks but needs to be updated with workflow metadata too
```

Comment:
Makes me wonder if these fields should be part of `Version` rather than `VersionMetadata`.

Reply:
> Ouch, so we we've been missing other stuff too. 

Yeah.

> Makes me wonder if these fields should be part of Version rather than VersionMetadata.

It's worth a thought. The initial reason `VersionMetadata` exists is that sometimes we want to change stuff for a frozen version. DOIs are an obvious example, we need to snapshot/freeze a version before issuing a DOI, but once we have a DOI we need to store it somewhere. 

Engine versions and descriptor type versions are less obvious, but maybe they're here because we computed them for pre-existing workflow versions

Reply:
> Engine versions and descriptor type versions are less obvious, but maybe they're here because we computed them for pre-existing workflow versions

Yes, it was for pre-existing versions. Plus if we ever tweak the algorithm in the future, which we actually might, see #5918, although that would be changing the language version, not the engine version.